docs/core/diagnostics/observability-with-otel.md

```diff
@@ -0,0 +1,425 @@
+
+### 1. Create the project
+
+Create a simple web API project by using the **ASP.NET Core Empty** template in Visual Studio or the following .NET CLI command:
+
+``` shell
+dotnet new web
+```
+
+
+### 2. Add metrics and activity definitions
+
+The following code defines a new metric (`greetings.count`) for the number of times the API has been called, and a new activity source (`OtPrGrYa.Sample`).
+
+:::code language="csharp" source="snippets/OTel-Prometheus-Graphana-Yaeger/csharp/Program.cs" id="Snippet_CustomMetrics":::
+
+### 3. Create an API endpoint
+
+:::code language="csharp" source="snippets/OTel-Prometheus-Graphana-Yaeger/csharp/Program.cs" id="Snippet_MapGet":::
+
+:::code language="csharp" source="snippets/OTel-Prometheus-Graphana-Yaeger/csharp/Program.cs" id="Snippet_SendGreeting":::
+
+> [!Note]
+> The API definition does not use anything specific to OpenTelemetry. It uses the .NET APIs for observability.
+
+### 4. Reference the OpenTelemetry packages
+
+Use the NuGet Package Manager or command line to add the following NuGet packages:
+
+:::code language="xml" source="snippets/OTel-Prometheus-Graphana-Yaeger/csharp/OTel-Prometheus-Graphana-Yaeger.csproj" id="Snippet_References":::
+
+> [!Note]
+> Use the latest versions, as the OTel APIs are constantly evolving.
+
+### 5. Configure OpenTelemetry with the correct providers
+
+:::code language="csharp" source="snippets/OTel-Prometheus-Graphana-Yaeger/csharp/Program.cs" id="Snippet_OTEL":::
+
+This code uses ASP.NET Core instrumentation to get metrics and activities from ASP.NET Core. It also registers the `Metrics` and `ActivitySource` providers for metrics and tracing respectively.
+
+The code uses the Prometheus exporter for metrics, which uses ASP.NET Core to host the endpoint, so you also need to add:
+
+:::code language="csharp" source="snippets/OTel-Prometheus-Graphana-Yaeger/csharp/Program.cs" id="Snippet_Prometheus":::
+
+### 6. Run the project
+
+Run the project and then access the API with the browser or curl.
+
+``` shell
+curl -k http://localhost:7275
+```
+
+Each time you request the page, it will increment the count for the number of greetings that have been made. You can access the metrics endpoint using the same base url, with the path `/metrics`. For example:
+
+#### 6.1 Log output
+
+The logging statements from the code are output using `ILogger`. By default, the [Console Provider](../extensions/logging.md?tabs=command-line#configure-logging) is enabled so that output is directed to the console. OTel in .NET doesn't include any logging exporters as logging is traditionally handled well by libraries such as [Serilog](https://serilog.net/) or [NLog](https://nlog-project.org/). Also, `stdout` and `stderr` output is redirected to log files by container systems such as [Kubernetes](https://kubernetes.io/docs/concepts/cluster-administration/logging/#how-nodes-handle-container-logs).
```

Comment:
> logging is traditionally handled well by libraries such as [Serilog](https://serilog.net/) or [NLog](https://nlog-project.org/)

Loggin in OTel .NET is handled via ILogger. This sentence reads like it is recommending Serilog, NLog, instead of ILogger - is that the intention?

Reply:
I thought that they plug into I logger, rather than being a replacement?

Is there a good logging source I should be using here for OTel - I feel its getting short changed, and would be happy to show it being exported via OTEL. It is shown with App Insights, but would be good to show with an OSS source too.

Reply:
OTLP + ElasticSearch
OTLP + Loki
are 2 good OSS choices to demo logging.

Reply:
my WIP demo that includes Loki for Logging:
https://github.com/cijothomas/opentelemetry-dotnet/tree/cijothomas/loki/docs/metrics/exemplars

(It does sent logs to Loki via OTLP+Collector . Did not have time to update the docs and polish it!)